### PR TITLE
Added new step for /var/lib/pgsql as a prerequisite

### PR DESCRIPTION
**Pull request by bangelic**
Per SATDOC-660, I added a step in Before You Begin in the Upgrading Satellite Server section. The step added is "If you previously upgraded Satellite from 6.7 or an earlier version, or if the /var/lib/pgsql directory has any content, ensure it is empty before proceeding. Satellite 6.11’s migrated postgres database content will be stored in this directory."

Cherry-pick into:

* [YES] Foreman 3.2
* [YES] Foreman 3.1
* For Foreman 3.0 or older, file a separate PR request

### CODE REVIEW

**Comment by ekohl** [comment]
> [YES] Foreman 3.2

The PR template uses task lists. See https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists#creating-task-lists for how to use them.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc [hunk 1]
```diff
@@ -19,6 +19,7 @@ Cloning is the recommended method because that prevents them being overwritten i
 To confirm if a template has been edited, you can view its *History* before you upgrade or view the changes in the audit log after an upgrade.
 In the {ProjectWebUI}, navigate to *Monitor* > *Audits* and search for the template to see a record of changes made.
 If you use the export method, restore your changes by comparing the exported template and the default template, manually applying your changes.
+* If you previously upgraded {Project} from 6.7 or an earlier version, or if the `/var/lib/pgsql` directory has any content, ensure it is empty before proceeding. {Project} {ProjectVersionPrevious}'s migrated `postgres` database content will be stored in this directory.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:22]: `6.7` is Satellite-only. Please check if this file is only included in builds for Satellite. If it also shows for Foreman, please either remove the version number or encapsulate in within `ifdef::satellite[]` and add an appropriate sentence for Foreman please.

Also, please stick to one sentence per line.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc [hunk 2]
```diff
@@ -19,6 +19,12 @@ Cloning is the recommended method because that prevents them being overwritten i
 To confirm if a template has been edited, you can view its *History* before you upgrade or view the changes in the audit log after an upgrade.
 In the {ProjectWebUI}, navigate to *Monitor* > *Audits* and search for the template to see a record of changes made.
 If you use the export method, restore your changes by comparing the exported template and the default template, manually applying your changes.
+ifdef::satellite[]
+* If you previously upgraded {Project} from an earlier version, or if the `/var/lib/pgsql` directory has any content, ensure it is empty before proceeding.
+endif::[]
+ifdef::satellite[]
+* {Project} {ProjectVersionPrevious}'s migrated `postgres` database content will be stored in this directory.
```
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:26]:
```suggestion
* {Project} {ProjectVersionPrevious}'s migrated `postgres` database content is stored in this directory.
```
Use the present tense at all times, as per IBM style.

**Review by melcorr — CHANGES_REQUESTED** [review]
Requesting some additional changes here @bangelic 
Thanks for your work on this.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc [hunk 3]
```diff
@@ -19,6 +19,10 @@ Cloning is the recommended method because that prevents them being overwritten i
 To confirm if a template has been edited, you can view its *History* before you upgrade or view the changes in the audit log after an upgrade.
 In the {ProjectWebUI}, navigate to *Monitor* > *Audits* and search for the template to see a record of changes made.
 If you use the export method, restore your changes by comparing the exported template and the default template, manually applying your changes.
+ifdef::satellite[]
+* If you previously upgraded {Project} from an earlier version, or if the `/var/lib/pgsql` directory has any content, ensure it is empty before proceeding.
```
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:23]: `or` feels odd to me. Isn't it `and`? Or rather: if `/var/lib/pgsql` has any content (as is the case when upgrading from older versions), empty it.

**Review by ekohl — COMMENTED** [review]
:+1: from me but I'd like to leave the final approval to @melcorr since she requested changes before.

**Review by melcorr — COMMENTED** [review]
One additional suggestion. Thanks for applying the other changes!

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc [hunk 4]
```diff
@@ -19,6 +19,10 @@ Cloning is the recommended method because that prevents them being overwritten i
 To confirm if a template has been edited, you can view its *History* before you upgrade or view the changes in the audit log after an upgrade.
 In the {ProjectWebUI}, navigate to *Monitor* > *Audits* and search for the template to see a record of changes made.
 If you use the export method, restore your changes by comparing the exported template and the default template, manually applying your changes.
+ifdef::satellite[]
+* If you previously upgraded {Project} from an earlier version, and the `/var/lib/pgsql` directory contains any content, as it does when upgrading from older versions, empty it before proceeding.
```
[the remark this answers is not in the document]
Comment (bangelic) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:23]: Made the change and will re commit. Thank you!

**Comment by bangelic** [comment]
@melcorr Is this good to be merged?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc [hunk 5]
```diff
@@ -19,6 +19,10 @@ Cloning is the recommended method because that prevents them being overwritten i
 To confirm if a template has been edited, you can view its *History* before you upgrade or view the changes in the audit log after an upgrade.
 In the {ProjectWebUI}, navigate to *Monitor* > *Audits* and search for the template to see a record of changes made.
 If you use the export method, restore your changes by comparing the exported template and the default template, manually applying your changes.
+ifdef::satellite[]
```
Comment (evgeni) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:22]: It's not specific to Satellite.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc [hunk 6]
```diff
@@ -19,6 +19,10 @@ Cloning is the recommended method because that prevents them being overwritten i
...
+* If you previously upgraded {Project} from an earlier version, and the `/var/lib/pgsql` directory contains any content, empty `/var/lib/pgsql` before proceeding.
+* {Project} {ProjectVersionPrevious}'s migrated `postgres` database content is stored in this directory.
```
Comment (evgeni) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:24]: This reads as if `/var/lib/pgsql` contains the *current* database content, which it does not. It does contain some old, not-cleaned-up content. I'd probably either completely remove this sentence, or reword it to something like: `/var/lib/pgsql` contained the PostgreSQL database content before the migration from PostgreSQL 9 to PostgreSQL 12 from the SCL.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc [hunk 7]
```diff
@@ -14,6 +14,8 @@ Cloning is the recommended method because that prevents them being overwritten i
 To confirm if a template has been edited, you can view its *History* before you upgrade or view the changes in the audit log after an upgrade.
 In the {ProjectWebUI}, navigate to *Monitor* > *Audits* and search for the template to see a record of changes made.
 If you use the export method, restore your changes by comparing the exported template and the default template, manually applying your changes.
+* If you previously upgraded {Project} from an earlier version, and the `/var/lib/pgsql` contained the PostgreSQL database content before the migration from PostgreSQL 9 to PostgreSQL 12 from the SCL, empty `/var/lib/pgsql` before proceeding.
+* {Project} {ProjectVersionPrevious}'s migrated `postgres` database content is stored in this directory.
```
Comment (evgeni) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:18]: This line can be removed now, as the content was merged into the previous sentence.

**Review by evgeni — CHANGES_REQUESTED** [review]

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc (hunk 7 above)
Comment (bangelic) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc]: Done and committed.

**Comment by bangelic** [comment]
@evgeni Removed the lines you suggested and recommitted.

**Comment by evgeni** [comment]
@bangelic the diff seems empty now?

**Comment by bangelic** [comment]
> @bangelic the diff seems empty now?

@evgeni Fixed it.

**Comment by evgeni** [comment]
Thanks!

One thing that I didn't realize in the previous review: currently this is placed under the general upgrade prereqs, but it's really only important for the EL7→EL8 Leapp path. Would you mind moving it?

**Comment by bangelic** [comment]
@evgeni So you want me to move **If you previously upgraded {Project} from an earlier version, and the `/var/lib/pgsql` contained the PostgreSQL database content before the migration from PostgreSQL 9 to PostgreSQL 12 from the SCL, empty `/var/lib/pgsql` before proceeding.** into the section we created for upgrading from RHEL 7 to RHEL 8 via LEAPP?

**Comment by bangelic** [comment]
> Exactly

@evgeni Done!

**Review by evgeni — APPROVED** [review]

**Comment by evgeni** [comment]
merged and picked (also to 3.3, which was missing from the original description):

   04d9fbd8..8a1cee93  3.1 -> 3.1
   0614aefe..d6222b3f  3.2 -> 3.2
   f010d77a..414f0520  3.3 -> 3.3
